Checkout/Checkout/Source/Validation/Validators/CardValidator.swift

```diff
@@ -178,63 +203,69 @@ public class CardValidator: CardValidating {
   // MARK: - Private
 
   private func validateExpiry(
-    expiryMonth: String,
-    expiryYear: String
+    expiryMonth: Int,
+    expiryYear: Int
   ) -> Result<ExpiryDate, ValidationError.ExpiryDate> {
-    guard let month = Int(expiryMonth) else {
-      return .failure(.invalidMonthString)
-    }
-
-    guard let year = Int(expiryYear) else {
-      return .failure(.invalidYearString)
-    }
-
-    guard month >= 1 && month <= 12 else {
+    guard expiryMonth >= 1 && expiryMonth <= 12 else {
       return .failure(.invalidMonth)
     }
 
-    let fourDigitYear: Int
-
-    switch self.fourDigitYear(
-      from: expiryYear,
-      year: year) {
-    case .success(let fourDigitYearResult):
-      fourDigitYear = fourDigitYearResult
-    case .failure(let expiryDateError):
-      return .failure(expiryDateError)
-    }
-
     let currentDate = Date()
+    // using UTC-12 as this is the latest timezone where a card could be valid
+    let cardExpiryComponents = DateComponents(
+      timeZone: .utcMinus12,
+      year: expiryYear,
+      month: expiryMonth
+    )
 
-    guard let providedDate = calendar.date(
-      from: DateComponents(
-        year: fourDigitYear,
-        month: month
-      )
-    ), providedDate >= currentDate else {
+    guard
+      let cardExpiryDate = calendar.date(from: cardExpiryComponents),
+      let nextMonth = calendar.date(byAdding: .month, value: 1, to: cardExpiryDate),
+      nextMonth > currentDate
+    else {
```

Comment:
I think having the first guard statement inlined with guard, as well as the else should be pretty common layout for the code. If you're concerned of too long lines, SwiftLint has some rule for maximum number of characters in a line.

Reply:
Was copied over, lets handle this in a future task